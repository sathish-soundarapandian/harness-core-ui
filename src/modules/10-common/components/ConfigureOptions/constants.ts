/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export enum ALLOWED_VALUES_TYPE {
  TIME = 'TIME',
  URL = 'URL',
  TEXT = 'TEXT',
  MULTI_SELECT = 'MULTI_SELECT',
  NUMBER = 'NUMBER'
}

export enum PLG_CD_GET_STARTED_VARIANTS {
  INFO_HEAVY = 'INFO_HEAVY',
  OLD_GET_STARTED = 'OLD_GET_STARTED'
}
