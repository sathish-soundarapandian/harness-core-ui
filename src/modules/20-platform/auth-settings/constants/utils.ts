/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { windowLocationUrlPartBeforeHash } from 'framework/utils/WindowLocation'

export const getSamlEndpoint = (accountId: string): string => {
  let url = `${windowLocationUrlPartBeforeHash()?.replace('/ng/', '')}`

  if (window.apiUrl) {
    if (window.apiUrl.startsWith('/')) {
      url = `${url}${window.apiUrl}`
    } else {
      url = window.apiUrl
    }
  }

  return `${url}/api/users/saml-login?accountId=${accountId}`
}
