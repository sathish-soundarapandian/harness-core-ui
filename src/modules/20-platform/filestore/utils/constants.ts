/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */
import { SORT_TYPE } from '@filestore/interfaces/FileStore'

export const FILE_STORE_ROOT = 'Root'
export const SEARCH_FILES = 'SEARCH'

export enum ExtensionType {
  YAML = 'yaml',
  JSON = 'json',
  TEXT = 'txt',
  BASH = 'sh',
  POWER_SHELL = 'ps',
  YML = 'yml',
  TPL = 'tpl',
  PY = 'py'
}

export enum LanguageType {
  YAML = 'yaml',
  JSON = 'json',
  TEXT = 'plaintext',
  BASH = 'shell',
  POWER_SHELL = 'powershell',
  YML = 'yaml',
  TPL = 'tpl'
}

export enum FSErrosType {
  UNSUPPORTED_FORMAT = 'UNSUPPORTED_FORMAT',
  FILE_USAGE = 'FILE_USAGE',
  DELETED_NODE = 'DELETED_NODE'
}

export enum FileStoreActionTypes {
  CREATE_NODE = 'CREATE_NODE',
  DELETE_NODE = 'DELETE_NODE',
  UPDATE_NODE = 'UPDATE_NODE',
  UPLOAD_NODE = 'UPLOAD_NODE',
  SORT_NODE = 'SORT_NODE'
}

export enum SELECT_FILES_TYPE {
  FILE_STORE = 'fileStore',
  ENCRYPTED = 'encrypted'
}

export const defaultSortItems = [
  SORT_TYPE.ALPHABETICAL,
  SORT_TYPE.ALPHABETICAL_DESC,
  SORT_TYPE.LAST_UPDATED,
  SORT_TYPE.LAST_UPDATED_DESC,
  SORT_TYPE.ALPHABETICAL_FILE_TYPE,
  SORT_TYPE.ALPHABETICAL_FOLDER_TYPE
]
