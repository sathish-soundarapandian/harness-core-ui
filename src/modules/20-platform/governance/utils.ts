/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { GovernanceMetadata } from 'services/cd-ng'

interface DoesGovernanceMetaDataHasErrorOrWarningRtn {
  governanceMetaDataHasError: boolean
  governanceMetaDataHasWarning: boolean
}
export const doesGovernanceHasErrorOrWarning = (
  governanceMetadata: GovernanceMetadata | undefined
): DoesGovernanceMetaDataHasErrorOrWarningRtn => {
  let governanceMetaDataHasWarning = false
  let governanceMetaDataHasError = false
  if (governanceMetadata && governanceMetadata?.status === 'error') {
    governanceMetaDataHasError = true
  }
  if (governanceMetadata && governanceMetadata?.status === 'warning') {
    governanceMetaDataHasWarning = true
  }
  return { governanceMetaDataHasError, governanceMetaDataHasWarning }
}
