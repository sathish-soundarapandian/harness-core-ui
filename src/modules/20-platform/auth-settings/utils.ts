/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { get } from 'lodash-es'
import type { CreateIpAllowlistConfigProps, UpdateIpAllowlistConfigProps } from '@harnessio/react-ng-manager-client'

import type { IIPAllowlistForm } from '@auth-settings/interfaces/IPAllowlistInterface'
import { SourceType } from '@auth-settings/interfaces/IPAllowlistInterface'
import { CurrencyType } from '@common/constants/SubscriptionTypes'

export function getAmountInCurrency(currency: CurrencyType, amount: number): string {
  switch (currency) {
    case CurrencyType.USD: {
      return amount.toLocaleString('en-US', {
        style: 'currency',
        currency: 'USD'
      })
    }
  }

  return ''
}

// dividing the actual amount by 100 as amount numbers are in cents
export function getDollarAmount(amount?: number, isYearly?: boolean): number {
  if (isYearly) {
    return (amount || 0) / 1200
  }
  return (amount || 0) / 100
}

export const getErrorMessage = (error: any): string =>
  get(error, 'data.error', get(error, 'data.message', error?.message))

export const buildCreateIPAllowlistPayload = (data: IIPAllowlistForm): CreateIpAllowlistConfigProps => {
  const allowSourceType = []
  if (data.allowSourceTypeUI) {
    allowSourceType.push(SourceType.UI)
  }
  if (data.allowSourceTypeAPI) {
    allowSourceType.push(SourceType.API)
  }
  return {
    body: {
      ip_allowlist_config: {
        allowed_source_type: allowSourceType,
        description: data?.description,
        enabled: false, // always false while creating
        identifier: data.identifier,
        ip_address: data.ipAddress,
        name: data.name,
        tags: data?.tags
      }
    }
  }
}

export const buildUpdateIPAllowlistPayload = (
  data: IIPAllowlistForm,
  overrideFields?: any
): UpdateIpAllowlistConfigProps => {
  const allowSourceType = []
  if (data.allowSourceTypeUI) {
    allowSourceType.push(SourceType.UI)
  }
  if (data.allowSourceTypeAPI) {
    allowSourceType.push(SourceType.API)
  }
  return {
    'ip-config-identifier': data.identifier,
    body: {
      ip_allowlist_config: {
        allowed_source_type: allowSourceType,
        description: data?.description,
        enabled: data?.enabled,
        identifier: data.identifier,
        ip_address: data.ipAddress,
        name: data.name,
        tags: data?.tags,
        ...overrideFields
      }
    }
  }
}
