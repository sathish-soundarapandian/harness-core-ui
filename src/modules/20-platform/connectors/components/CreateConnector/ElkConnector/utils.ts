/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { ElkAuthType } from '@connectors/constants'
import type { FormData } from '@connectors/interfaces/ConnectorInterface'
import type { ConnectorConfigDTO } from 'services/cd-ng'

type InitializeElkConnectorArgs = {
  prevStepData?: ConnectorConfigDTO
  accountId: string
  projectIdentifier?: string
  orgIdentifier?: string
}

export function initializeElkConnector({
  prevStepData,
  accountId,
  projectIdentifier,
  orgIdentifier
}: InitializeElkConnectorArgs): FormData {
  const defaultObj = {
    url: '',
    accountId,
    username: '',
    authType: ElkAuthType.USERNAME_PASSWORD,
    password: undefined,
    projectIdentifier,
    orgIdentifier
  }

  if (!prevStepData) {
    return defaultObj
  }
  const { spec, ...prevData } = prevStepData
  return {
    ...defaultObj,
    url: prevData?.url || spec?.controllerUrl || spec?.url || '',
    password: prevData?.password || spec?.passwordRef,
    clientSecretRef: prevData?.clientSecretRef || spec?.clientSecretRef,
    authType: prevData?.authType || spec?.authType || ElkAuthType.USERNAME_PASSWORD,
    username: prevData?.username || spec?.username || '',
    clientId: prevData?.clientId || spec?.clientId,
    apiKeyId: prevData?.apiKeyId || spec?.apiKeyId,
    apiKeyRef: prevData?.apiKeyRef || spec?.apiKeyRef
  }
}
