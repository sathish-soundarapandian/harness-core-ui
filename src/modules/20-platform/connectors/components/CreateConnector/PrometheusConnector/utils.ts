/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { ValueType } from '@secrets/components/TextReference/TextReference'
import { setSecretField } from '@secrets/utils/SecretField'
import type { ConnectorConfigDTO, GetSecretV2QueryParams } from 'services/cd-ng'
import type { CustomHealthConnectorDTO, CustomHealthKeyAndValue } from 'services/cv'
import type {
  BaseCompFields,
  EncryptedEntity,
  PlainEntity
} from '../CustomHealthConnector/components/CustomHealthHeadersAndParams/CustomHealthHeadersAndParams.types'
import { DefaultHeadersInitialValues } from './CreatePrometheusConnector.constants'

async function transformSpecHeader(
  scopeQueryParams: GetSecretV2QueryParams,
  entity?: CustomHealthConnectorDTO['headers']
): Promise<BaseCompFields['headers']> {
  if (!entity?.length) {
    return DefaultHeadersInitialValues
  }

  const transformedEntities: BaseCompFields['headers'] = []
  const encryptedEntities = entity.filter(e => {
    if (!e.valueEncrypted) {
      transformedEntities.push({
        key: e.key,
        value: {
          textField: e.value,
          fieldType: ValueType.TEXT
        }
      } as PlainEntity)
    }
    return e.valueEncrypted
  })

  await Promise.all(
    encryptedEntities.map(encryptedEntity =>
      setSecretField(encryptedEntity.encryptedValueRef as string, scopeQueryParams)
    )
  )
    .then(secretObjs => {
      encryptedEntities.forEach((e, index) =>
        transformedEntities.push({
          key: e.key,
          value: {
            fieldType: ValueType.ENCRYPTED,
            secretField: secretObjs[index]
          }
        } as EncryptedEntity)
      )
    })
    .catch(() => ({}))

  return transformedEntities
}

export async function initializePrometheusConnectorWithStepData(
  prevStepData?: ConnectorConfigDTO | null,
  scopeQueryParams?: GetSecretV2QueryParams
): Promise<ConnectorConfigDTO | undefined> {
  if (!prevStepData) {
    return
  }

  const { spec, ...prevData } = prevStepData
  const updatedInitialValues = {
    ...spec,
    ...prevData
  }

  if (prevData?.url) {
    return {
      ...updatedInitialValues,
      url: prevData?.url,
      headers: prevData?.headers,
      username: prevData?.username,
      passwordRef: prevData?.passwordRef
    }
  }

  return {
    ...updatedInitialValues,
    url: spec.url,
    headers: await transformSpecHeader(scopeQueryParams as GetSecretV2QueryParams, spec.headers),
    username: spec?.username,
    passwordRef: await setSecretField(spec?.passwordRef, scopeQueryParams as GetSecretV2QueryParams)
  }
}

function getSpecHeaderAndEntity(entity: PlainEntity | EncryptedEntity): CustomHealthKeyAndValue | null {
  if (entity?.value?.fieldType === ValueType.ENCRYPTED && entity.key && entity.value.secretField?.referenceString) {
    return {
      valueEncrypted: true,
      encryptedValueRef: entity.value.secretField.referenceString,
      key: entity.key
    }
  } else if (entity?.value?.fieldType === ValueType.TEXT && entity.key && entity.value.textField) {
    return {
      valueEncrypted: false,
      value: entity.value.textField,
      key: entity.key
    }
  }

  return null
}

export function transformStepHeadersAndParamsForPayloadForPrometheus(headers: BaseCompFields['headers']): {
  headers: CustomHealthConnectorDTO['headers']
} {
  const apiHeaders: CustomHealthConnectorDTO['headers'] = []

  headers?.forEach(header => {
    const transformedHeader = getSpecHeaderAndEntity(header)
    if (transformedHeader) {
      apiHeaders.push(transformedHeader)
    }
  })

  return {
    headers: apiHeaders
  }
}
