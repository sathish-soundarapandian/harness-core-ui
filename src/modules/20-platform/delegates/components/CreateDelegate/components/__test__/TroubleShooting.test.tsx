/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { fireEvent, render, act } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { DelegateTypes } from '@delegates/constants'
import TroubleShooting from '../DelegateInstallationError/TroubleShooting'

describe('Create Common Problems Tab', () => {
  test('render data', () => {
    const { container } = render(
      <TestWrapper>
        <TroubleShooting />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('show event error btn', () => {
    const { container } = render(
      <TestWrapper>
        <TroubleShooting />
      </TestWrapper>
    )
    const noBtn = container.querySelector('.noBtn')
    fireEvent.click(noBtn!)
    expect(container).toMatchSnapshot()
  })

  test('Show docker specific troubleshooting guide', () => {
    const { getByText, getByRole } = render(
      <TestWrapper>
        <TroubleShooting delegateType={DelegateTypes.DOCKER} />
      </TestWrapper>
    )
    expect(getByText('delegates.delegateNotInstalled.verifyStatus2')).toBeInTheDocument()
    const yesBtn = getByRole('button', { name: 'yes' })
    act(() => {
      fireEvent.click(yesBtn!)
    })
    expect(getByText('delegates.delegateNotInstalled.verifyLogs1')).toBeInTheDocument()
  })

  test('Click back button after verifying running status of Docker Container', () => {
    const { getByText, getByRole } = render(
      <TestWrapper>
        <TroubleShooting delegateType={DelegateTypes.DOCKER} />
      </TestWrapper>
    )
    expect(getByText('delegates.delegateNotInstalled.verifyStatus2')).toBeInTheDocument()
    const yesBtn = getByRole('button', { name: 'yes' })
    act(() => {
      fireEvent.click(yesBtn!)
    })
    expect(getByText('delegates.delegateNotInstalled.verifyLogs1')).toBeInTheDocument()
    const backBtn = getByRole('button', { name: 'back' })
    act(() => {
      fireEvent.click(backBtn!)
    })
    expect(getByText('delegates.delegateNotInstalled.verifyStatus2')).toBeInTheDocument()
  })

  test('Verify troubleshoot guide when docker container is not running', () => {
    const { container, getByText } = render(
      <TestWrapper>
        <TroubleShooting delegateType={DelegateTypes.DOCKER} />
      </TestWrapper>
    )
    expect(getByText('delegates.delegateNotInstalled.verifyStatus2')).toBeInTheDocument()
    const noBtn = container.querySelector('.noBtn')
    act(() => {
      fireEvent.click(noBtn!)
    })
    expect(getByText('delegates.delegateNotInstalled.heartbeatFailed')).toBeInTheDocument()
  })

  test('Click on restart button when docker container is not running', () => {
    const { container, getByText, getByRole } = render(
      <TestWrapper>
        <TroubleShooting delegateType={DelegateTypes.DOCKER} />
      </TestWrapper>
    )
    expect(getByText('delegates.delegateNotInstalled.verifyStatus2')).toBeInTheDocument()
    const noBtn = container.querySelector('.noBtn')
    act(() => {
      fireEvent.click(noBtn!)
    })
    expect(getByText('delegates.delegateNotInstalled.heartbeatFailed')).toBeInTheDocument()
    const restart = getByRole('button', { name: 'restart' })
    act(() => {
      fireEvent.click(restart!)
    })
    expect(getByText('delegates.delegateNotInstalled.containerRunning')).toBeInTheDocument()
  })

  test('Render K8s troubleshoot guide when delegate type is not passed', () => {
    const { getByText, getByRole } = render(
      <TestWrapper>
        <TroubleShooting />
      </TestWrapper>
    )
    expect(getByText('delegates.delegateNotInstalled.podCommand')).toBeInTheDocument()
    const yesBtn = getByRole('button', { name: 'yes' })
    act(() => {
      fireEvent.click(yesBtn!)
    })
    expect(getByText('delegates.delegateNotInstalled.harnessErrorValidation')).toBeInTheDocument()
  })
  test('Render K8s troubleshoot guide and verify guide when pod does not come up ', () => {
    const { container, getByText } = render(
      <TestWrapper>
        <TroubleShooting />
      </TestWrapper>
    )
    expect(getByText('delegates.delegateNotInstalled.podCommand')).toBeInTheDocument()
    const noBtn = container.querySelector('.noBtn')
    act(() => {
      fireEvent.click(noBtn!)
    })
    expect(getByText('delegates.delegateNotInstalled.eventErrorBlock')).toBeInTheDocument()
  })
  test('Render K8s troubleshoot guide and verify cluster permission steps when pod does not come up ', () => {
    const { container, getByText, getByRole } = render(
      <TestWrapper>
        <TroubleShooting />
      </TestWrapper>
    )
    expect(getByText('delegates.delegateNotInstalled.podCommand')).toBeInTheDocument()
    const noBtn = container.querySelector('.noBtn')
    act(() => {
      fireEvent.click(noBtn!)
    })
    expect(getByText('delegates.delegateNotInstalled.eventErrorBlock')).toBeInTheDocument()
    const verifyClusterPermissionBtn = getByRole('button', {
      name: 'delegates.delegateNotInstalled.permissionError warning-sign'
    })
    act(() => {
      fireEvent.click(verifyClusterPermissionBtn!)
    })
    expect(getByText('delegates.delegateNotInstalled.checkClusterPermission')).toBeInTheDocument()
  })
  test('Render K8s troubleshoot guide and verify subsequent steps when pod does not come up ', () => {
    const { container, getByText, getByRole } = render(
      <TestWrapper>
        <TroubleShooting />
      </TestWrapper>
    )
    expect(getByText('delegates.delegateNotInstalled.podCommand')).toBeInTheDocument()
    const noBtn = container.querySelector('.noBtn')
    act(() => {
      fireEvent.click(noBtn!)
    })
    expect(getByText('delegates.delegateNotInstalled.eventErrorBlock')).toBeInTheDocument()
    const verifyClusterPermissionBtn = getByRole('button', {
      name: 'delegates.delegateNotInstalled.permissionError warning-sign'
    })
    act(() => {
      fireEvent.click(verifyClusterPermissionBtn!)
    })
    expect(getByText('delegates.delegateNotInstalled.checkClusterPermission')).toBeInTheDocument()

    const backBtn = getByRole('button', { name: 'back' })

    act(() => {
      fireEvent.click(backBtn!)
    })

    act(() => {
      fireEvent.click(noBtn!)
    })

    const containerFailureBtn = getByRole('button', {
      name: 'delegates.delegateNotInstalled.crashloopError warning-sign'
    })
    act(() => {
      fireEvent.click(containerFailureBtn!)
    })

    expect(getByText('delegates.delegateNotInstalled.eventErrorBlock')).toBeInTheDocument()
  })
})
