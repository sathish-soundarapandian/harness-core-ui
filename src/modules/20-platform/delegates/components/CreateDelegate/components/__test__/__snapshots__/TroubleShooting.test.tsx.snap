// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Common Problems Tab render data 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main"
    style="width: 630px;"
  >
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex"
      style="border-bottom: 0.5px solid #B0B1C4;"
    >
      <button
        aria-label="back"
        class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--without-shadow"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          back
        </span>
      </button>
      <p
        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black StyledProps--font-size-medium StyledProps--font-weight-bold"
        style="line-height: 22px; display: flex; align-items: center;"
      >
        <span
          class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-blue800"
          data-icon="info"
        >
          <svg
            height="16"
            viewBox="0 0 12 12"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M0 6a6 6 0 1012 0A6 6 0 000 6zm.75 0a5.25 5.25 0 1010.5 0A5.25 5.25 0 00.75 6zm4.206 3.104v.646h2.682v-.646l-.836-.144V4.203H4.956v.65l.836.144V8.96zm1.846-7.352v1.03h-1.01v-1.03z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        delegates.delegateNotInstalled.tabs.commonProblems.hereIsWhatYouCanDo
      </p>
      <button
        aria-label="restart"
        class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--without-shadow"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          restart
        </span>
      </button>
    </div>
    <div
      class="Layout--vertical StyledProps--main"
      style="width: 511px;"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black100"
        >
          delegates.delegateNotInstalled.statusOfCluster
        </p>
        <a
          href="https://docs.harness.io/"
          rel="noreferrer"
          style="text-align: end; font-size: 13px;"
          target="_blank"
        >
          delegates.delegateNotInstalled.tabs.commonProblems.troubleshoot
        </a>
      </div>
      <div
        class="StyledProps--font StyledProps--main podCmndField StyledProps--intent-primary StyledProps--padding-small StyledProps--font-align-center StyledProps--flex"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          delegates.delegateNotInstalled.podCommand
        </p>
        <span
          class="bp3-popover-wrapper"
        >
          <span
            class="bp3-popover-target"
          >
            <div
              class=""
              tabindex="0"
            >
              <span
                class="bp3-icon StyledProps--main copyIcon"
                data-icon="copy-alt"
              >
                <svg
                  fill="none"
                  height="20"
                  viewBox="0 0 8 9"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M7.273 1.403c0-.109-.073-.181-.182-.181H3.453c-.108 0-.181.072-.181.18v.181a.363.363 0 01-.727 0v-.361c0-.397.328-.722.728-.722h4c.4 0 .727.325.728.721v5.42a.726.726 0 01-.728.722h-1.09a.364.364 0 01-.365-.36c0-.2.165-.362.364-.362h.91c.108 0 .18-.072.18-.18V1.403zM0 7.847V2.958c0-.359.337-.652.748-.652h4.114c.411 0 .749.293.749.652v4.89c0 .359-.338.652-.749.652H.748C.337 8.5 0 8.207 0 7.847z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </span>
        </span>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main podCmndVerification"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black100"
      >
        delegates.delegateNotInstalled.podComeUp
      </p>
      <div
        class="StyledProps--font StyledProps--main"
      >
        <button
          aria-label="yes"
          class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            yes
          </span>
        </button>
        <button
          aria-label="no"
          class="bp3-button Button--button StyledProps--font StyledProps--main noBtn Button--with-current-color"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            no
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Create Common Problems Tab show event error btn 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main"
    style="width: 630px;"
  >
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex"
      style="border-bottom: 0.5px solid #B0B1C4;"
    >
      <button
        aria-label="back"
        class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--without-shadow"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          back
        </span>
      </button>
      <p
        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black StyledProps--font-size-medium StyledProps--font-weight-bold"
        style="line-height: 22px; display: flex; align-items: center;"
      >
        <span
          class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-blue800"
          data-icon="info"
        >
          <svg
            height="16"
            viewBox="0 0 12 12"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M0 6a6 6 0 1012 0A6 6 0 000 6zm.75 0a5.25 5.25 0 1010.5 0A5.25 5.25 0 00.75 6zm4.206 3.104v.646h2.682v-.646l-.836-.144V4.203H4.956v.65l.836.144V8.96zm1.846-7.352v1.03h-1.01v-1.03z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        delegates.delegateNotInstalled.tabs.commonProblems.hereIsWhatYouCanDo
      </p>
      <button
        aria-label="restart"
        class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--without-shadow"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          restart
        </span>
      </button>
    </div>
    <div
      class="Layout--vertical StyledProps--main"
      style="width: 511px;"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black100"
        >
          delegates.delegateNotInstalled.statusOfCluster
        </p>
        <a
          href="https://docs.harness.io/"
          rel="noreferrer"
          style="text-align: end; font-size: 13px;"
          target="_blank"
        >
          delegates.delegateNotInstalled.tabs.commonProblems.troubleshoot
        </a>
      </div>
      <div
        class="StyledProps--font StyledProps--main podCmndField StyledProps--intent-primary StyledProps--padding-small StyledProps--font-align-center StyledProps--flex"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          delegates.delegateNotInstalled.podCommand
        </p>
        <span
          class="bp3-popover-wrapper"
        >
          <span
            class="bp3-popover-target"
          >
            <div
              class=""
              tabindex="0"
            >
              <span
                class="bp3-icon StyledProps--main copyIcon"
                data-icon="copy-alt"
              >
                <svg
                  fill="none"
                  height="20"
                  viewBox="0 0 8 9"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M7.273 1.403c0-.109-.073-.181-.182-.181H3.453c-.108 0-.181.072-.181.18v.181a.363.363 0 01-.727 0v-.361c0-.397.328-.722.728-.722h4c.4 0 .727.325.728.721v5.42a.726.726 0 01-.728.722h-1.09a.364.364 0 01-.365-.36c0-.2.165-.362.364-.362h.91c.108 0 .18-.072.18-.18V1.403zM0 7.847V2.958c0-.359.337-.652.748-.652h4.114c.411 0 .749.293.749.652v4.89c0 .359-.338.652-.749.652H.748C.337 8.5 0 8.207 0 7.847z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </span>
        </span>
      </div>
    </div>
    <div
      class="Layout--vertical StyledProps--main"
      style="width: 511px;"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black100"
      >
        delegates.delegateNotInstalled.eventErrorBlock
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main btnContainer"
    >
      <button
        class="bp3-button Button--button StyledProps--font StyledProps--main error Button--with-current-color Button--iconOnly Button--withRightIcon"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          delegates.delegateNotInstalled.permissionError
        </span>
        <span
          class="bp3-icon bp3-icon-warning-sign StyledProps--main StyledProps--color StyledProps--color-orange400"
          icon="warning-sign"
        >
          <svg
            data-icon="warning-sign"
            height="10"
            viewBox="0 0 16 16"
            width="10"
          >
            <desc>
              warning-sign
            </desc>
            <path
              d="M15.84 13.5l.01-.01-7-12-.01.01c-.17-.3-.48-.5-.85-.5s-.67.2-.85.5l-.01-.01-7 12 .01.01c-.09.15-.15.31-.15.5 0 .55.45 1 1 1h14c.55 0 1-.45 1-1 0-.19-.06-.35-.15-.5zm-6.85-.51h-2v-2h2v2zm0-3h-2v-5h2v5z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
      <button
        class="bp3-button Button--button StyledProps--font StyledProps--main error addSpacing Button--with-current-color Button--iconOnly Button--withRightIcon"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          delegates.delegateNotInstalled.crashloopError
        </span>
        <span
          class="bp3-icon bp3-icon-warning-sign StyledProps--main StyledProps--color StyledProps--color-orange400"
          icon="warning-sign"
        >
          <svg
            data-icon="warning-sign"
            height="10"
            viewBox="0 0 16 16"
            width="10"
          >
            <desc>
              warning-sign
            </desc>
            <path
              d="M15.84 13.5l.01-.01-7-12-.01.01c-.17-.3-.48-.5-.85-.5s-.67.2-.85.5l-.01-.01-7 12 .01.01c-.09.15-.15.31-.15.5 0 .55.45 1 1 1h14c.55 0 1-.45 1-1 0-.19-.06-.35-.15-.5zm-6.85-.51h-2v-2h2v2zm0-3h-2v-5h2v5z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
      <button
        class="bp3-button Button--button StyledProps--font StyledProps--main error addSpacing Button--with-current-color Button--iconOnly Button--withRightIcon"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          delegates.delegateNotInstalled.pullError
        </span>
        <span
          class="bp3-icon bp3-icon-warning-sign StyledProps--main StyledProps--color StyledProps--color-orange400"
          icon="warning-sign"
        >
          <svg
            data-icon="warning-sign"
            height="10"
            viewBox="0 0 16 16"
            width="10"
          >
            <desc>
              warning-sign
            </desc>
            <path
              d="M15.84 13.5l.01-.01-7-12-.01.01c-.17-.3-.48-.5-.85-.5s-.67.2-.85.5l-.01-.01-7 12 .01.01c-.09.15-.15.31-.15.5 0 .55.45 1 1 1h14c.55 0 1-.45 1-1 0-.19-.06-.35-.15-.5zm-6.85-.51h-2v-2h2v2zm0-3h-2v-5h2v5z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
